src/hmem_cuda.c

```diff
@@ -754,11 +726,34 @@ int cuda_host_unregister(void *ptr)
 
 enum cuda_xfer_setting cuda_get_xfer_setting(void)
 {
+	enum cuda_xfer_setting hmem_cuda_xfer_setting;
+	int ret, param_value = 1;
+
+	/* Fetch the value of the environment variable FI_HMEM_CUDA_ENABLE_XFER */
+	ret = fi_param_get_bool(NULL, "hmem_cuda_enable_xfer", &param_value);
+	if (ret == -EINVAL){
+		fprintf(stderr, "FI_HMEM_CUDA_ENABLE_XFER was set to an invalid value by the user."
+			" FI_HMEM_CUDA_ENABLE_XFER is boolean and can be set to only 0/false/no/off or"
+			" 1/true/yes/on.\n");
+		abort();
+	}
+
+	if (ret) {
+		hmem_cuda_xfer_setting = CUDA_XFER_UNSPECIFIED;
+	} else {
+		hmem_cuda_xfer_setting = (param_value > 0) ? CUDA_XFER_ENABLED : CUDA_XFER_DISABLED;
+	}
+
+	FI_INFO(&core_prov, FI_LOG_CORE, "cuda_xfer_setting: %d\n", hmem_cuda_xfer_setting);
+
 	return hmem_cuda_xfer_setting;
 }
 
 bool cuda_is_ipc_enabled(void)
 {
+	/* CUDA IPC is only enabled if cuda_hmem_init() is successful */
+	cuda_ipc_enabled = cuda_hmem_devices_verified == true;
```

Comment:
I do not think we need the global variable `cuda_ipc_enabled` any more.

Reply:
is it enough? I do not think ipc_enabled can be true is `hmem_cuda_enable_xfer` is false?